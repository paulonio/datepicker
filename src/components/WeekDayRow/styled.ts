import styled from 'styled-components';

export const WeekWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #333333;
  font-size: 14px;
  font-weight: 700;
  line-height: 19px;
`;

export const WeekDay = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
`;
