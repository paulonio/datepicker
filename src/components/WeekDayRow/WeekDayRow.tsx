import React, { FC } from 'react';

import { MONDAY_STARTS, SUNDAY_STARTS } from '@constants/constants';
import calendar from '@utils/ClassCalendar';
import type { Mode, WeekendStatus } from '@/types/types';

import { WeekDay, WeekWrapper } from './styled';

interface WeekDayRowProps {
  mode: Mode;
  weekendStatus: WeekendStatus;
}

const WeekDayRow: FC<WeekDayRowProps> = ({ mode, weekendStatus }) => {
  const row = mode === 'su' ? SUNDAY_STARTS : MONDAY_STARTS;

  return (
    <WeekWrapper>
      {row.map((day) => {
        const showWeekend = calendar.showWeekendTitle(day, weekendStatus);
        return (
          <WeekDay key={day} showWeekend={showWeekend} data-testid="week-day">
            {day}
          </WeekDay>
        );
      })}
    </WeekWrapper>
  );
};

export default WeekDayRow;
