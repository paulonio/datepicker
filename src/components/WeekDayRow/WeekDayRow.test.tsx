import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import WeekDayRow from '@components/WeekDayRow/WeekDayRow';

describe('WeekDayRow', () => {
  it('should render WeekDayRow component and starts on monday', () => {
    const { container } = render(<WeekDayRow mode="mo" />);

    expect(container).toMatchSnapshot();
    const firstWeekDay = screen.getAllByTestId('week-day')[0];
    expect(firstWeekDay).toHaveTextContent('Mo');
  });

  it('should render WeekDayRow component and starts on sunday', () => {
    render(<WeekDayRow mode="su" />);

    const firstWeekDay = screen.getAllByTestId('week-day')[0];
    expect(firstWeekDay).toHaveTextContent('Su');
  });
});
