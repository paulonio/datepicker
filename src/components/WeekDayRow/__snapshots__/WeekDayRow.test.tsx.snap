// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekDayRow should render WeekDayRow component and starts on monday 1`] = `
<div>
  <div
    class="sc-bgqQcB MEEiE"
  >
    <div
      class="sc-gTRrQi kXvLIr"
      data-testid="week-day"
    >
      Mo
    </div>
    <div
      class="sc-gTRrQi kXvLIr"
      data-testid="week-day"
    >
      Tu
    </div>
    <div
      class="sc-gTRrQi kXvLIr"
      data-testid="week-day"
    >
      We
    </div>
    <div
      class="sc-gTRrQi kXvLIr"
      data-testid="week-day"
    >
      Th
    </div>
    <div
      class="sc-gTRrQi kXvLIr"
      data-testid="week-day"
    >
      Fr
    </div>
    <div
      class="sc-gTRrQi kXvLIr"
      data-testid="week-day"
    >
      Sa
    </div>
    <div
      class="sc-gTRrQi kXvLIr"
      data-testid="week-day"
    >
      Su
    </div>
  </div>
</div>
`;
